src/main/java/org/broadinstitute/hellbender/utils/hdf5/HDF5PoNCreator.java

```diff
@@ -0,0 +1,712 @@
+
+        try (final HDF5File ponReader = new HDF5File(inputHDF5Filename, HDF5File.OpenMode.READ_ONLY)) {
+            final PoN inputPoN = new HDF5PoN(ponReader);
+            final ReadCountCollection logNormalizedCounts = new ReadCountCollection(SetUniqueList.setUniqueList(inputPoN.getPanelTargets()), SetUniqueList.setUniqueList(new ArrayList<>(inputPoN.getPanelSampleNames())), inputPoN.getLogNormalizedCounts());
+            final ReadCountCollection coverageProfile = new ReadCountCollection(SetUniqueList.setUniqueList(inputPoN.getTargets()), SetUniqueList.setUniqueList(new ArrayList<>(inputPoN.getSampleNames())), inputPoN.getNormalizedCounts());
+
+            final ReductionResult newReduction = calculateReducedPanelAndPInverses(logNormalizedCounts, newNumberOfEigenSamples, logger, ctx);
+
+            final PoN newPoN = new RamPoN(inputPoN, newReduction);
+            writeTargetFactorNormalizeReadCountsAndTargetFactors(outputHDF5Filename, coverageProfile, newPoN.getTargetFactors().getColumn(0), inputPoN.getRawTargets());
+            writeLogNormalsReducedPanel(outputHDF5Filename, logNormalizedCounts, newReduction, ctx);
+        }
+    }
+
+    /**
+     * SVD and Pseudo inverse calculation.
+     *
+     * @param logNormals the input counts for the SVD and reduction steps, fully normalized and already logged.
+     * @param requestedNumberOfEigenSamples user requested number of eigen samples for the reduced panel.
+     * @return never {@code null}.
+     */
+    @VisibleForTesting
+    static ReductionResult calculateReducedPanelAndPInverses(final ReadCountCollection logNormals, final OptionalInt requestedNumberOfEigenSamples, final Logger logger, final JavaSparkContext ctx) {
+
+        if (ctx == null) {
+            logger.warn("No Spark context provided, not going to use Spark...");
+        }
+
+        final RealMatrix logNormalCounts = logNormals.counts();
+        final int numberOfCountColumns = logNormalCounts.getColumnDimension();
+
+        logger.info("Starting the SVD decomposition of the log-normal counts ...");
+        final long svdStartTime = System.currentTimeMillis();
+        final SVD logNormalsSVD = SVDFactory.createSVD(logNormals.counts(), ctx);
+        final long svdEndTime = System.currentTimeMillis();
+        logger.info(String.format("Finished the SVD decomposition of the log-normal counts. Elapse of %d seconds", (svdEndTime - svdStartTime) / 1000));
+
+        final int numberOfEigenSamples = determineNumberOfEigenSamples(requestedNumberOfEigenSamples, numberOfCountColumns, logNormalsSVD, logger);
+        logger.info(String.format("Including %d eigen samples in the reduced PoN", numberOfEigenSamples));
+        final RealMatrix logNormalsV = logNormalsSVD.getV();
+        final RealMatrix logNormalsEigenV = logNormalsV.getSubMatrix(0, numberOfCountColumns - 1, 0, numberOfEigenSamples - 1);
+
+        final RealMatrix reducedCounts = logNormalCounts.multiply(logNormalsEigenV);
```

Comment:
This is slower than it needs to be.  The SVD of matrix M is M = U*sigma*V' (using MATLAB notation for transpose) and we are calculating `reducedCounts` as M*V, where V is restricted to E eigensamples.  By orthogonality of V, V'V is the ExE identity, with zeroes below.  Thus sigma*V'*V equals sigma truncated to the first E columns.  Since sigma is rectangular diagonal, the result U*sigma*V'*V is thus a matrix whose e-th column is the e-th column of U times the e-th diagonal entry of sigma.